language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/diagnostic/DiagnosticsHelper.java

```diff
@@ -115,6 +117,9 @@ public synchronized void schedulePublishDiagnostics(ExtendedLanguageClient clien
      * @param context LS context
      */
     public synchronized void compileAndSendDiagnostics(ExtendedLanguageClient client, DocumentServiceContext context) {
+        if (context.languageServercontext().get(LanguageServerLightWeightMode.class) != null) {
```

Comment:
Let's add this check to the PublishDiagnoticSubscriber instead https://github.com/ballerina-platform/ballerina-lang/blob/8ccdc067648a303b81e43aa4c67d9a2e96bcfcae/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/eventsync/subscribers/PublishDiagnosticSubscriber.java#L45